CMakeLists.txt

```diff
@@ -1,273 +1,229 @@
-    set( CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -Xcompiler -fPIC -std=c++11 -D_FORCE_INLINES --expt-extended-lambda" )
-    if ( CCACHE_FOUND )
-        set( CMAKE_CUDA_COMPILER_LAUNCHER "${CCACHE_FOUND}" )
-    endif ()
-else ()
-    message( STATUS "Building Knowhere CPU version" )
-endif ()
-
-if ( KNOWHERE_SUPPORT_SPTAG )
-    message( STATUS "Building Knowhere with SPTAG supported" )
-    add_compile_definitions( "KNOWHERE_SUPPORT_SPTAG" )
-endif ()
-
-if ( CMAKE_BUILD_TYPE STREQUAL "Release" )
-    append_flags( CMAKE_CXX_FLAGS FLAGS "-O3" )
-    append_flags( CUDA_NVCC_FLAGS FLAGS "-O3" )
-else ()
-    append_flags( CMAKE_CXX_FLAGS FLAGS "-O0" "-g" )
-    append_flags( CUDA_NVCC_FLAGS FLAGS "-O0" "-g" )
-endif ()
-
-append_flags( CMAKE_CXX_FLAGS
-              FLAGS
-                    "-fPIC"
-                    "-DELPP_THREAD_SAFE"
-                    "-fopenmp"
-                    )
-
-# option "-rdynamic" is to print out detailed backtrace info for debug-ease
-if ( LINUX )
-    append_flags( CMAKE_CXX_FLAGS FLAGS "-Werror" "-rdynamic" )
-endif ()
-
-#*********************** Coding style check tools **********************
-find_package( Python COMPONENTS Interpreter Development )
-find_package( ClangToolsKw )
-set( BUILD_SUPPORT_DIR "${CMAKE_SOURCE_DIR}/build-support" )
-
-if ( "$ENV{CMAKE_EXPORT_COMPILE_COMMANDS}" STREQUAL "1" OR CLANG_TIDY_FOUND )
-  # Generate a Clang compile_commands.json "compilation database" file for use
-  # with various development tools, such as Vim's YouCompleteMe plugin.
-  # See http://clang.llvm.org/docs/JSONCompilationDatabase.html
-  set( CMAKE_EXPORT_COMPILE_COMMANDS ON )
-  set( CMAKE_CXX_STANDARD_INCLUDE_DIRECTORIES ${CMAKE_CXX_IMPLICIT_INCLUDE_DIRECTORIES} )
-endif ()
-
-#
-# "make lint" target
-#
-if ( NOT KNOWHERE_VERBOSE_LINT )
-    set( KNOWHERE_LINT_QUIET "--quiet" )
-endif ()
-
-if ( NOT LINT_EXCLUSIONS_FILE )
-    # source files matching a glob from a line in this file
-    # will be excluded from linting (cpplint, clang-tidy, clang-format)
-    set( LINT_EXCLUSIONS_FILE ${BUILD_SUPPORT_DIR}/lint_exclusions.txt )
-endif ()
-
-if ( NOT IGNORE_CHECKS_FILE )
-    set( IGNORE_CHECKS_FILE ${BUILD_SUPPORT_DIR}/ignore_checks.txt )
-endif ()
-
-find_program( CPPLINT_BIN NAMES cpplint cpplint.py HINTS ${BUILD_SUPPORT_DIR} )
-message( STATUS "Found cpplint executable at ${CPPLINT_BIN}" )
-
-#
-# "make lint-knowhere" targets
-#
-add_custom_target( lint-knowhere
-    ${PYTHON_EXECUTABLE}    ${BUILD_SUPPORT_DIR}/run_cpplint.py
-                            --cpplint_binary    ${CPPLINT_BIN}
-                            --exclude_globs     ${LINT_EXCLUSIONS_FILE}
-                            --source_dir        ${CMAKE_CURRENT_SOURCE_DIR}
-                            ${KNOWHERE_LINT_QUIET}
-                            )
-
-#
-# "make clang-format" and "make check-clang-format" targets
-#
-if ( ${CLANG_FORMAT_FOUND} )
-    # runs clang format and updates files in place.
```

Comment:
the original CMakeLists support clang_format, seems this function has been removed.

Reply:
next time, we will use git hook to check this.